### ISSUE TITLE
value for domain lovelace violates check constraint "lovelace_check"

### ISSUE THREAD

**Issue by mmahut** [input-output-hk/cardano-db-sync#351]
On the mainnet, just after slot height 4830407, cardano-db-sync is stuck with following:

```
Oct 17 12:02:43  g71fl8m54rmr7f59qilbrv2r4b389d4n-cardano-db-sync[9243]: [db-sync-node:Info:215] [2020-10-17 12:02:43.56 UTC] Starting chainSyncClient
Oct 17 12:02:43  g71fl8m54rmr7f59qilbrv2r4b389d4n-cardano-db-sync[9243]: [db-sync-node:Info:215] [2020-10-17 12:02:43.64 UTC] Cardano.Db tip is at slot 11369248, block 4830407
Oct 17 12:02:43  g71fl8m54rmr7f59qilbrv2r4b389d4n-cardano-db-sync[9243]: [db-sync-node:Info:220] [2020-10-17 12:02:43.64 UTC] Running DB thread
Oct 17 12:02:43  g71fl8m54rmr7f59qilbrv2r4b389d4n-cardano-db-sync[9243]: [db-sync-node:Info:220] [2020-10-17 12:02:43.65 UTC] Rolling back to slot 11369248, hash c847d4ffc835132dcfc07c3ff1e10c8090cdaaf9c05bea5a0b9132470aec64ba
Oct 17 12:02:43  g71fl8m54rmr7f59qilbrv2r4b389d4n-cardano-db-sync[9243]: [db-sync-node:Info:220] [2020-10-17 12:02:43.65 UTC] Deleting blocks numbered: []
Oct 17 12:02:43  g71fl8m54rmr7f59qilbrv2r4b389d4n-cardano-db-sync[9243]: [db-sync-node:Error:220] [2020-10-17 12:02:43.68 UTC] Bad pledge amount: 9999372036855000000 > maxLovelace. See https://github.com/input-output-hk/cardano-ledger-specs/issues/1551
Oct 17 12:02:43  g71fl8m54rmr7f59qilbrv2r4b389d4n-cardano-db-sync[9243]: [db-sync-node:Error:220] [2020-10-17 12:02:43.68 UTC] runDBThread: DbInsertException "PoolUpdate" (SqlError {sqlState = "23514", sqlExecStatus = FatalError, sqlErrorMsg = "value for domain lovelace violates check constraint \"lovelace_check\"", sqlErrorDetail = "", sqlErrorHint = ""})
Oct 17 12:02:43  g71fl8m54rmr7f59qilbrv2r4b389d4n-cardano-db-sync[9243]: [db-sync-node:Error:215] [2020-10-17 12:02:43.68 UTC] ChainSyncWithBlocksPtcl: DbInsertException "PoolUpdate" (SqlError {sqlState = "23514", sqlExecStatus = FatalError, sqlErrorMsg = "value for domain lovelace violates check constraint \"lovelace_check\"", sqlErrorDetail = "", sqlErrorHint = ""})
Oct 17 12:02:43  g71fl8m54rmr7f59qilbrv2r4b389d4n-cardano-db-sync[9243]: [db-sync-node.Subscription:Error:211] [2020-10-17 12:02:43.68 UTC] [String "Application Exception: LocalAddress {getFilePath = \"/run/cardano-node/node.socket\"} DbInsertException \"PoolUpdate\" (SqlError {sqlState = \"23514\", sqlExecStatus = FatalError, sqlErrorMsg = \"value for domain lovelace violates check constraint \\\"lovelace_check\\\"\", sqlErrorDetail = \"\", sqlErrorHint = \"\"})",String "SubscriptionTrace"]
Oct 17 12:02:43  g71fl8m54rmr7f59qilbrv2r4b389d4n-cardano-db-sync[9243]: [db-sync-node.ErrorPolicy:Error:4] [2020-10-17 12:02:43.68 UTC] [String "ErrorPolicyUnhandledApplicationException (DbInsertException \"PoolUpdate\" (SqlError {sqlState = \"23514\", sqlExecStatus = FatalError, sqlErrorMsg = \"value for domain lovelace violates check constraint \\\"lovelace_check\\\"\", sqlErrorDetail = \"\", sqlErrorHint = \"\"}))",String "ErrorPolicyTrace",String "LocalAddress {getFilePath = \"/run/cardano-node/node.socket\"}"]
```

As a workaround, removing the constraint check seems to be a workaround, however, I'm sure it was in place for a good reason.

```psql> ALTER DOMAIN lovelace DROP CONSTRAINT lovelace_check;```

**Comment by erikd** [comment]
:cry: Yeah, this is disappointing. It looks like `pool_update.fixedCost` amount can be `> maxLovelace`. I thought the ledger would guard against this nonsensical error but sisnce it does not guard against `pool_update.pledge` being sane (https://github.com/input-output-hk/cardano-ledger-specs/issues/1551) perhaps this is not surprising.

**Comment by erikd** [comment]
Now we get a warning:
```
Bad fixed cost amount: 9223372036855000000 > maxLovelace. See https://github.com/input-output-hk/cardano-db-sync/issues/351
```
and `db-sync` keeps on running (the bad value is warned about and inserted into the database).
